account_manager/src/validator/modify.rs

```diff
@@ -1,91 +1,42 @@
 pub const DISABLE: &str = "disable";
 
 pub const PUBKEY_FLAG: &str = "pubkey";
 pub const ALL: &str = "all";
 
-pub fn cli_app() -> Command {
-    Command::new(CMD)
-        .about("Modify validator status in validator_definitions.yml.")
-        .display_order(0)
-        .subcommand(
-            Command::new(ENABLE)
-                .about("Enable validator(s) in validator_definitions.yml.")
-                .arg(
-                    Arg::new(PUBKEY_FLAG)
-                        .long(PUBKEY_FLAG)
-                        .value_name("PUBKEY")
-                        .help("Validator pubkey to enable")
-                        .action(ArgAction::Set)
-                        .display_order(0),
-                )
-                .arg(
-                    Arg::new(ALL)
-                        .long(ALL)
-                        .help("Enable all validators in the validator directory")
-                        .action(ArgAction::SetTrue)
-                        .help_heading(FLAG_HEADER)
-                        .conflicts_with(PUBKEY_FLAG)
-                        .display_order(0),
-                ),
-        )
-        .subcommand(
-            Command::new(DISABLE)
-                .about("Disable validator(s) in validator_definitions.yml.")
-                .arg(
-                    Arg::new(PUBKEY_FLAG)
-                        .long(PUBKEY_FLAG)
-                        .value_name("PUBKEY")
-                        .help("Validator pubkey to disable")
-                        .action(ArgAction::Set)
-                        .display_order(0),
-                )
-                .arg(
-                    Arg::new(ALL)
-                        .long(ALL)
-                        .help("Disable all validators in the validator directory")
-                        .action(ArgAction::SetTrue)
-                        .help_heading(FLAG_HEADER)
-                        .conflicts_with(PUBKEY_FLAG)
-                        .display_order(0),
-                ),
-        )
-}
-
-pub fn cli_run(matches: &ArgMatches, validator_dir: PathBuf) -> Result<(), String> {
+pub fn cli_run(modify_config: &Modify, validator_dir: PathBuf) -> Result<(), String> {
     // `true` implies we are setting `validator_definition.enabled = true` and
     // vice versa.
-    let (enabled, sub_matches) = match matches.subcommand() {
-        Some((ENABLE, sub_matches)) => (true, sub_matches),
-        Some((DISABLE, sub_matches)) => (false, sub_matches),
-        Some((unknown, _)) => {
-            return Err(format!(
-                "{} does not have a {} command. See --help",
-                CMD, unknown
-            ))
-        }
-        _ => return Err(format!("No command provided for {}. See --help", CMD)),
+    let (enabled, sub_matches) = match modify_config {
+        Modify::Enable(sub_matches) => (true, Box::new(sub_matches) as Box<dyn Modifiable>),
+        Modify::Disable(sub_matches) => (false, Box::new(sub_matches) as Box<dyn Modifiable>),
```

Comment:
Unless I'm missing something I think we can remove both the `Box` and `Modifiable` if we just do: 

```suggestion
    let (enabled, pubkey, is_all) = match modify_config {
        Modify::Enable(enable) => (true, enable.pubkey.clone(), enable.all),
        Modify::Disable(disable) => (false, disable.pubkey.clone(), disable.all),
    };
```

Reply:
c0ce1cc
9cfcc59